File: android/app/src/main/java/it/axant/axemas/AXMSectionController.java
package it.axant.axemas;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.ViewGroup;
import java.lang.ref.WeakReference;

public class AXMSectionController {
    private WeakReference<SectionFragment> section;

    public AXMSectionController(SectionFragment section) {
        this.section = new WeakReference<SectionFragment>(section);
    }

    protected SectionFragment getSection() {
        return this.section.get();
    }

    public void sectionDidLoad() {
        // web page finished loading
    }

    public void sectionWillLoad() {
        // web page has not yet started to load
    }

    public void sectionOnViewCreate(ViewGroup view) {
        // axemas view/fragment got created
    }

    public boolean isInsideWebView(MotionEvent ev) {
        return true;
    }

    /*
    The following methods are specific to Android as they follow the android excluside lifecycle.
     */
    public void sectionFragmentWillPause() { }
    public void sectionFragmentWillResume() { }
    public void sectionFragmentOnSaveInstanceState(Bundle outState) { }
    public void sectionFragmentOnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) { }
    public void sectionFragmentOnActivityResult(int requestCode, int resultCode, Intent data) { }
    public void actionbarRightButtonAction() {}
}

File: winphone/framework/AXEMAS/AXMSectionController.cs
using axemas.Controls;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;

namespace axemas
{
    public class AXMSectionController {

        private WeakReference _section;
        protected SectionViewPage section {
            get {
                return _section != null ? _section.Target as SectionViewPage : null;
            }
        }

        public AXMSectionController()
        {
            this._section = null;
        }

        internal void bindToSection(SectionViewPage section)
        {
            this._section = new WeakReference(section);
        }

        public virtual void sectionDidLoad()
        {
            // web page finished loading
        }

        public virtual void sectionWillLoad()
        {
            // web page has not yet started to load
        }

        public virtual void sectionOnViewCreate(UIElement Content)
        {
            // axemas viewPage got created
        }

        public virtual void sectionViewPageWillPause() { }
        public virtual void sectionViewPageWillResume() { }
    }
}
